fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java

```diff
@@ -427,6 +430,29 @@ public static TStorageFormat analyzeStorageFormat(Map<String, String> properties
         }
     }
 
+    // analyzeCompressionType will parse the compression type from properties
+    public static TCompressionType analyzeCompressionType(Map<String, String> properties) throws AnalysisException {
+        String compressionType;
+        if (properties != null && properties.containsKey(PROPERTIES_COMPRESSION)) {
+            compressionType = properties.get(PROPERTIES_COMPRESSION);
+            properties.remove(PROPERTIES_COMPRESSION);
+        } else {
+            return TCompressionType.LZ4_FRAME;
+        }
+
+        if (compressionType.equalsIgnoreCase("lz4")) {
```

Comment:
you can use CompressionUtils directly.

Reply:
OK, but in fact, not all compression algorithm in TCompressionType can be used for data pages by now. Comparison between different compression algorithm is unavoidable. This must be done in PropertyAnalyzer or in CompressionUtils. Of course, we can do it in CompressionUtils.

Reply:
compressionutils is a common function, if storage does not supported some compression type, it is the storage to do the compression check. But storage can use TCompressionType after CompressionUtils converting.

Reply:
I think it is a good idea to do converting/checking in CompressionUtils before the storage use the compression type. Because, if the user exec the CREATE TABLE statement with a unsupported compression type, we should throw the error in analyzeCompressionType immediately.